shell/common/surface.cc

```diff
@@ -20,8 +28,21 @@ bool SurfaceFrame::Submit() {
   return submitted_;
 }
 
-Surface::Surface() = default;
+SkCanvas* SurfaceFrame::SkiaCanvas() {
+  return surface_ != nullptr ? surface_->getCanvas() : nullptr;
+}
 
-Surface::~Surface() = default;
+bool SurfaceFrame::PerformSubmit() {
+  if (submit_callback_ == nullptr) {
```

Comment:
What is the rationale for clearing surface_ in the "true" case, and leaving it as it is in both of the "false" cases?